modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -39,18 +55,46 @@ export default function CartQuickAdd() {
 	const paginatorTotalCountRef = useRef(0);
 	const requestAbortControllerRef = useRef(new AbortController());
 
-	const [formattedProducts, setFormattedProducts] = useState([]);
-
-	const [productsQuery, setProductsQuery] = useState('');
-	const [productsWithOptions, setProductsWithOptions] = useState([]);
+	const [availableSKUs, setAvailableSKUs] = useState([]);
+	const [formattedSKUs, setFormattedSKUs] = useState([]);
 	const [quantityError, setQuantityError] = useState(false);
 	const [quickAddToCartError, setQuickAddToCartError] = useState(false);
-	const [selectedProducts, setSelectedProducts] = useState([]);
+	const [searchInputValue, setSearchInputValue] = useState('');
+	const [selectedSKUs, setSelectedSKUs] = useState([]);
 
 	const {cartItems = [], channel} = cartState;
 	const accountId = cartState.accountId;
 	const channelId = channel.channel.id;
 
+	useEffect(() => {
+		const productsApiURL = new URL(
+			`${themeDisplay.getPathContext()}${CHANNEL_RESOURCE_ENDPOINT}/${channelId}/products?accountId=${accountId}&nestedFields=skus&pageSize=-1&skus.accountId=${accountId}`,
```

Comment:
`pageSize=-1` is wrong. @smottal did changes in order to make a search instead of getting all the products in the system. Please add your development on top of @smottal changes.

Reply:
I need all products, how to search for a replacement product otherwise? Can I specify an `id` to the URL to get the specific product?

Reply:
I have access to the `replacementSkuId` and I'm looping over products to find the product/sku with a matching `id`, is there a way for me to fetch that specific product/sku?

Reply:
Why do you need all products? Can't we save the replacementSkuId in order to understand which one to add to the cart?

Reply:
How do I find out which SKU the `replacementSkuId` belongs to?

I have access to `replacementSkuId`, but [addToCart](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/data.js#L28) expects an SKU object with `skuId`, `skuOptions` and `quantity` properties.

I cannot just send a string representing the id of the SKU to the addToCart function and expect it to work, I need to send the entire SKU object.

@alessiorendina  How would you propose I get the SKU where `SKU.id === replacementSkuId` without looping over the entire list of SKUs?

Reply:
@alessiorendina can you check my comments above please, thanks!